dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala

```diff
@@ -75,79 +74,82 @@ class CaffeLoader[T: ClassTag](prototxtPath: String, modelPath: String,
   }
 
   private def getBlob(name: String, ind: Int): Option[Caffe.BlobProto] = {
-    if (name2LayerV2.contains(name) && name2LayerV2(name).getBlobsCount != 0) {
+    if (name2LayerV2.contains(name) && name2LayerV2(name).getBlobsCount > ind) {
       Some(name2LayerV2(name).getBlobs(ind))
-    } else if (name2LayerV1.contains(name) && name2LayerV1(name).getBlobsCount != 0) {
+    } else if (name2LayerV1.contains(name) && name2LayerV1(name).getBlobsCount > ind) {
       Some(name2LayerV1(name).getBlobs(ind))
     } else {
       None
     }
   }
 
-  private def loadParameters(name: String, destPara: Array[Tensor[T]]):
-  (Tensor[T], Tensor[T]) = {
+  private def loadParameters(name: String, params: Table): Unit = {
     logger.info(s"load parameters for $name ...")
     val caffeWeight = getBlob(name, 0)
-    if (caffeWeight.isEmpty) return (null, null)
-    val weightList = caffeWeight.get.getDataList
-    require(destPara != null && destPara(0).nElement() == weightList.size(),
-      s"weight element number is not equal between caffe layer and bigdl module $name, " +
-        s"data shape in caffe is ${ caffeWeight.get.getShape() }," +
-        s" while data shape in bigdl is ${ destPara(0).size().mkString(",") }")
-    require(destPara(0).isContiguous())
-    val weightData = destPara(0).storage().array()
-    for (i <- 0 until weightList.size()) {
-      weightData(i) = ev.fromType[Float](weightList.get(i))
+    if (caffeWeight.isDefined) {
+      require(params.contains("weight"), s"$name should contain weight")
+      val caffeWeightData = caffeWeight.get.getDataList
+      val weight = params[Tensor[T]]("weight")
```

Comment:
if caffe layer contains weight, then weight is required in bigdl module, as shown in line 90